homeassistant/components/http/auth.py

```diff
@@ -67,7 +68,7 @@ def async_sign_path(
         secret,
         algorithm="HS256",
     )
-    return f"{path}?{SIGN_QUERY_PARAM}={encoded}"
+    return str(URL(path)%f"{SIGN_QUERY_PARAM}={encoded}")
```

Comment:
Feels like this should be: `str(URL(path) % {SIGN_QUERY_PARAM: unencoded})`

Reply:
YARL has full support for updating query params. 

However, we need to encode the query params and make sure that those are exactly as-is when the request comes in.